src/loader/bgw_launcher.c

```diff
@@ -314,10 +319,22 @@ start_db_schedulers(HTAB *db_htab)
 	if (!bgw_total_workers_increment_by(ndatabases))
 	{
 		ereport(LOG, (errcode(ERRCODE_CONFIGURATION_LIMIT_EXCEEDED),
-					  errmsg("total databases = %ld TimescaleDB background worker limit %d", hash_get_num_entries(db_htab), guc_max_background_workers),
+					  errmsg("total databases = %ld TimescaleDB background worker limit %d, so no schedulers allocated", hash_get_num_entries(db_htab), guc_max_background_workers),
 					  errhint("You may start background workers manually by using the  _timescaledb_internal.start_background_workers() function in each database you would like to have a scheduler worker in.")));
 		return;
 	}
+	else
+	{
+		/*
+		 * Mark every scheduler as incremented. This is true regardless of
+		 * what happens below.
+		 */
+		hash_seq_init(&hash_seq, db_htab);
+		while ((current_entry = hash_seq_search(&hash_seq)) != NULL)
```

Comment:
This exact loop happens also below. Why not set this boolean in that loop instead of looping twice?